unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BIWSaveController.cs

```diff
@@ -2,26 +2,31 @@
 using DCL.Controllers;
 using UnityEngine;
 
-public class BIWSaveController : BIWController
+public interface IBIWSaveController
 {
-    [SerializeField]
-    private float msBetweenSaves = 5000f;
+    public void SetSaveActivation(bool isActive, bool tryToSave = false);
+    public void TryToSave();
+    public void ForceSave();
+}
 
-    [Header("Prefab reference")]
-    public BuilderInWorldBridge builderInWorldBridge;
+public class BIWSaveController : BIWController, IBIWSaveController
+{
+    private const float MS_BETWEEN_SAVES = 5000f;
 
     public int numberOfSaves { get; private set; } = 0;
 
+    private BuilderInWorldBridge bridge;
+
     private float nextTimeToSave;
     private bool canActivateSave = true;
 
-    public override void Init()
+    public override void Init(BIWContext biwContext)
     {
-        base.Init();
+        base.Init(biwContext);
 
-        builderInWorldBridge = InitialSceneReferences.i.builderInWorldBridge;
-        if (builderInWorldBridge != null)
-            builderInWorldBridge.OnKernelUpdated += TryToSave;
+        bridge = InitialSceneReferences.i.builderInWorldBridge;
+        if (bridge != null)
+            bridge.OnKernelUpdated += TryToSave;
```

Comment:
I think this relationship is backwards. Bridges call features, not the other way around. Anyway, I think this is not on the scope of this PR.

Reply:
They way it was designed is that the bridge is only a way to communicate between kernel <> Unity, it only has that responsibility to act as a bridge, It is weird to me that a bridge needs to know when it should say to the `SaveController ` "Save now" since that responsibility is on the `SaveController `

Let me know what do you think